packages/core/nuxt-theme-module/theme/pages/Category.vue

```diff
@@ -387,23 +387,28 @@ export default {
       return category?.label || items[0].label;
     });
 
+    const selectedFilters = ref({});
+    const setSelectedFilters = () => {
+      if (!facets.value.length || Object.keys(selectedFilters.value).length) return;
+      selectedFilters.value = facets.value.reduce((prev, curr) => ({
+        ...prev,
+        [curr.id]: curr.options
+          .filter(o => o.selected)
+          .map(o => o.id)
+      }), {});
+    };
+
     onSSR(async () => {
       await search(th.getFacetsFromURL());
+      setSelectedFilters();
     });
 
     const { changeFilters, isFacetColor } = useUiHelpers();
     const { toggleFilterSidebar } = useUiState();
-    const selectedFilters = ref({});
 
     onMounted(() => {
       context.root.$scrollTo(context.root.$el, 2000);
-      if (!facets.value.length) return;
-      selectedFilters.value = facets.value.reduce((prev, curr) => ({
-        ...prev,
-        [curr.id]: curr.options
-          .filter(o => o.selected)
-          .map(o => o.id)
-      }), {});
+      setSelectedFilters();
```

Comment:
I think here it would be reasonable to have an `if` so that if the filters are already selected it should not fire this method.

Reply:
The `if` condition is inside the `setSelectedFilters` method. Or is it not what you mean?

```
if (!facets.value.length || Object.keys(selectedFilters.value).length) return;
```

Reply:
I havent noticed the second argument. I approve then